### ISSUE TITLE
Request: Ability to search "Comment" column of Memory Viewer

### ISSUE THREAD

**Issue by dsasmblr** [cheat-engine/cheat-engine#513]
![comment](https://user-images.githubusercontent.com/22781064/52159624-e0239180-2674-11e9-841b-559bf9f7b668.png)

Is there a way for CE to perhaps index all comments found such that said index could be searchable? There are many times in games where multiple addresses contain the same keyword(s) of interest that no instructions appear to be accessing.

While it might seem like there's a reason for instructions not accessing an address, there are cases where I verify that an instruction in the Memory Viewer IS continuously reading an address that contains a string like what's pictured above. If I then add that address to the address list, then attempt to view instructions accessing the address, nothing shows up. This might be due to a misunderstanding on my behalf somehow.

Anyway, I think the general ability to search through comments (presumably after running the "Dissect Code" feature), would be supremely helpful!

**Comment by dsasmblr** [comment]
Question about those, as I do each frequently (well, I just start with running Dissect Code). There seems to be conditions where the comment will change, or that the string I see in a comment which I happen upon while scrolling through the disassembler in a subroutine of interest, didn't actually show up in the referenced strings result nor string search while scanning memory (including Unicode, read/write, etc.).

I could very well just have done things in the perfectly wrong order up to this point or something, lol. I feel like I'm taking crazy pills now... :S

**Comment by cheat-engine** [comment]
Is the code inside a module or runtime generated ?

**Comment by dsasmblr** [comment]
So, in the screenshot above, I noted that comment while I was scrolling down through the sub-routine I was in. It was an `LEA` instruction, so I found the address that was being accessed and went to it. Static address at `game.exe+offset`.

So I then scanned memory for that string and DID have that address as one of the results; however, when trying to right-click on that address to find instructions accessing it, nothing comes up. Yet I see that `LEA` instruction is reading the address continuously if I look for addresses that instruction is accessing. Maybe my debugger settings or something?

Lastly, I honestly can't say that I checked referenced strings after doing the aforementioned. This particular one may have been there and could have perhaps worked as a frame of reference, but I'm, like, 95% certain I've later seen string references in the disassembler that I didn't see initially after running Dissect Code and searching for said string references. That 5% doubt is hitting me strongly right now, though...lol.

**Comment by cheat-engine** [comment]
and "PostEffect" doesn't show up in referenced strings ?

Also, LEA does not access memory, it's just a calculation (e.g. `lea eax,[0]` is fine and doesn't crash) .  It's possible it's the start of a string array and rcx is an offset into that array which explains why the debugger doesn't show it's used

**Comment by dsasmblr** [comment]
I just checked, and it **does** show up! As for `LEA`, that makes total sense. I suppose I was thrown off by the fact that CE will show the address that's being calculated if looking for addresses the instruction accesses. Thus I thought the other way around would also work. Nice clarification.

I had what I thought was another issue, but I now realize the problem was something stupid I did. So I'll just go ahead and close this ticket and pretend this all never happened...lol. Thanks for your time and help, @mgrinzPlayer and @cheat-engine. =)
